### PR TITLE
[FEATURE] allow PageTSconfig option to filter for mappings

### PR DESCRIPTION
**Pull request by jpmschuler**
This will allow easier supplying of sitepackages, as you can filter
mappings by page branch easily. Most template extension will provide
mappings as well as PageTSconfig, so with this you can easily do a

mod.web_txtemplavoilaplusLayout.filterMaps = vendor/yourextension

to not show other mappings in that branch. Thus allowing
multi-template-setups easily, while still retaining the access check
and thus allowing editors for different templates to not mix them up,
as we can now filter by page tree as well as user.

As there might be extension supplying FCEs only (e.g. bootstrap style
containers). there is an option to define an array of allowed stuff
like

mod.web_txtemplavoilaplusLayout.filterMaps.1 = vendor/yourextension/Page
mod.web_txtemplavoilaplusLayout.filterMaps.2 = vendor/yourotherextension/FCE

### CODE REVIEW

File: Classes/Service/ItemsProcFunc.php [hunk 1]
```diff
@@ -81,4 +86,30 @@ protected function getScope(array $params)
...
+     * @param string $mappingPlace
+     *
+     * @return bool
+     */
+    protected static function checkIfMapIsFiltered(array $tvpPageTsConfig, string $mappingPlace): bool
+    {
+        if (isset($tvpPageTsConfig['filterMaps.'])) {
+            $allowedPlaces = $tvpPageTsConfig['filterMaps.'];
+        }
+        elseif (isset($tvpPageTsConfig['filterMaps'])) {
```
Comment (opi99) [inline Classes/Service/ItemsProcFunc.php:101]: The elseif or else should go one line above

File: Configuration/TSConfig/Page.ts [hunk 2]
```diff
@@ -4,3 +4,13 @@ mod.web_txtemplavoilaplusLayout.enableContentAccessWarning = 1
 mod.web_txtemplavoilaplusLayout.enableLocalizationLinkForFCEs = 0
 mod.web_txtemplavoilaplusLayout.useLiveWorkspaceForReferenceListUpdates = 1
 mod.web_txtemplavoilaplusLayout.adminOnlyPageStructureInheritance = fallback
+
+# example for filtering allowed maps allow single string or array
```
Comment (opi99) [inline Configuration/TSConfig/Page.ts:8]: This should go in Documentation and not here.
Maybe we should rename the Documentation folder to doc_old and create a new one and fill this up part by part

**Review by opi99 — CHANGES_REQUESTED** [review]

File: Classes/Service/ItemsProcFunc.php (hunk 1 above)
Comment (jpmschuler) [reply Classes/Service/ItemsProcFunc.php]: It seems you suggest a specific code style I tried some but couldn't make my PHPStorm not change files so I'm just guessing. 
Should perhaps code style rules be added to the project? I could create a PR for the default TYPO3 core php-cs-fixer rules.

File: Configuration/TSConfig/Page.ts (hunk 2 above)
Comment (jpmschuler) [reply Configuration/TSConfig/Page.ts:8]: Probably, yes. But this is rather out of scope of the ticket and PR, is it?

Comment (jpmschuler) [reply Configuration/TSConfig/Page.ts:8]: (and removing the info altogether before a new docs is there is probably not helpful)

Comment (jpmschuler) [reply Configuration/TSConfig/Page.ts]: done via https://github.com/pluspol-interactive/templavoilaplus/pull/343

**Review by opi99 — APPROVED** [review]

**Pull request merged**